app/api/routes/recommendations.py

```
@@ -40,3 +47,102 @@ def get_recommendations(
+    """
+    추천 논문 상호작용 데이터 저장.
+    
+    체류 시간, 스크롤 깊이, 북마크 등의 사용자 상호작용 데이터를
+    recommendation_interactions 컬렉션에 저장합니다.
+    """
+    # recommendation_id로부터 user_id와 paper_id 조회
+    recommendations_coll = db_mongo["paper_recommendations"]
+    rec = recommendations_coll.find_one({"_id": ObjectId(recommendation_id)})
+    
+    if not rec:
+        raise NotFoundException(f"Recommendation {recommendation_id} not found")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)